engines/misc/src/main/java/org/alfresco/transform/misc/transformers/TextToPdfContentTransformer.java

```diff
@@ -344,6 +403,137 @@ public PDDocument createPDFFromText(Reader text, int pageLimit)
+
+            if (name == null)
+            {
+                name = fontName != null ? fontName : getDefaultFont();
+            }
+
+            PDType1Font pdType1Font = PagedTextToPDF.getStandardFont(name);
+
+            if (pdType1Font != null)
+            {
+                font = pdType1Font;
+            }
+            else
+            {
+                FontMapping<TrueTypeFont> mapping = FontMappers.instance().getTrueTypeFont(name, null);
+
+                if (mapping != null && mapping.getFont() != null && !mapping.isFallback())
+                {
+                    try
+                    {
+                        font = PDType0Font.load(doc, mapping.getFont().getOriginalData());
+                    }
+                    catch (Exception e)
+                    {
+                        logger.error("Error loading mapping font: " + e.getMessage());
+                    }
+                }
+                else
+                {
+                    String nameWithExtension = name + ".ttf";
+
+                    FontFileFinder fontFileFinder = new FontFileFinder();
+                    List<URI> uris = fontFileFinder.find();
+
+                    for (URI uri : uris)
+                    {
+
+                        if (uri.getPath().contains(nameWithExtension))
+                        {
+                            InputStream fontIS = null;
+                            try
+                            {
+                                fontIS = new FileInputStream(new File(uri));
+                                if (null != fontIS)
+                                {
+                                    PDDocument documentMock = new PDDocument();
+                                    font = PDType0Font.load(documentMock, fontIS);
+                                    break;
+                                }
+                            }
+                            catch (IOException ioe)
+                            {
+                                String msg = "Error loading font " + name + " : " + ioe.getMessage();
+                                logger.error(msg, ioe);
+                            }
+                            finally
+                            {
+                                if (fontIS != null)
+                                {
+                                    try
+                                    {
+                                        fontIS.close();
+                                    }
+                                    catch (Exception e)
+                                    {
+                                        logger.error("Error closing font inputstream: " + e.getMessage());
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (font == null)
+            {
+                if (!name.equals(defaultFont))

```

Comment:
Is `defaultFont` the same as `getDefaultFont()` above? Is this the default font for the whole transformer or something specified by the PDF document?

Reply:
The main class `TextToPdfContentTransformer` has 2 class level attributes:

`String pdfBoxDefaultFont`
- Used to store the value from `transform.core.misc.pdfBox.defaultFont` property ([here](https://github.com/Alfresco/alfresco-transform-core/blob/1fa2ed033007e7ade468e594e169b735c154a373/engines/misc/src/main/java/org/alfresco/transform/misc/transformers/TextToPdfContentTransformer.java#L94))

`PagedTextToPDF transformer`
- Responsible for performing the transformation
- The `PagedTextToPDF` class also has a `defaultFont` attribute ([here](https://github.com/Alfresco/alfresco-transform-core/blob/1fa2ed033007e7ade468e594e169b735c154a373/engines/misc/src/main/java/org/alfresco/transform/misc/transformers/TextToPdfContentTransformer.java#L282)) that is initialized [here](https://github.com/Alfresco/alfresco-transform-core/blob/1fa2ed033007e7ade468e594e169b735c154a373/engines/misc/src/main/java/org/alfresco/transform/misc/transformers/TextToPdfContentTransformer.java#L109) - this way the transformer is aware of the configured default font, otherwise, it wouldn't be able to get the value from Spring boot property.